### PR TITLE
Museum page

### PR DESCRIPTION
**Pull request by CalMWolfs**
Requires https://github.com/NotEnoughUpdates/NotEnoughUpdates-REPO/pull/957
Only makes the museum API request once on the museum page to reduce the number of API calls. 

Due to the way divan's drills are in the museum (in special instead of rarities) it means that the 4 titanium can sometimes not show up even if they are donated) I can detect them if the divans drill is in the museum but most people do not keep it there.
Even if they have put the divan's drill in and then taken it out it still wont show. The only way is if they donated the x655 before getting a divans drill

I kept the 7 x 4 grid so that it aligns with the actual museum gui
- Summary of total museum value along with the number of items donated from each category.
- You can click on a particular item to see when it was donated and if it is currently in the user's museum
image

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/profileviewer/SkyblockProfiles.java [hunk 1]
```diff
@@ -484,6 +487,175 @@ public class SkyblockProfile {
...
+			private final Map<String, JsonArray> weaponItems = new HashMap<>();
+			private final Map<String, JsonArray> armorItems = new HashMap<>();
+			private final Map<String, JsonArray> raritiesItems = new HashMap<>();
+			private final List<JsonArray> specialItems = new ArrayList<>();
+			private final Map<String, Pair<Long, Boolean>> savedItems = new HashMap<>();
+
+			private MuseumData(JsonObject museumJson) {
+				JsonObject museum = Constants.MUSEUM;
+				if (museumJson == null || museumJson.isJsonNull() || museumJson.get("members").isJsonNull() || museum == null) {
+					museumValue = -2;
```
Comment (Lulonaut) [inline src/main/java/io/github/moulberry/notenoughupdates/profileviewer/SkyblockProfiles.java:504]: I think its more helpful to show the repo notification here `Utils.showOutdatedRepoNotification()`

**Review by Lulonaut — DISMISSED** [review]

**Review by jani270 — CHANGES_REQUESTED** [review]
image
image
😡

**Comment by NopoTheGamer** [comment]
It would be nice to have some sort of button that hides items you haven't submitted and only shows ones you have. Other than that LGTM

**Comment by CalMWolfs** [comment]
> It would be nice to have some sort of button that hides items you haven't submitted and only shows ones you have. Other than that LGTM

Is it all good if this gets done in a future pr if there is need for it as I'm pretty busy for the next few weeks

**Comment by NopoTheGamer** [comment]
> Is it all good if this gets done in a future pr if there is need for it as I'm pretty busy for the next few weeks
sure

**Review by Lulonaut — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]
unlucky efe

**Pull request merged**